### PR TITLE
i41: Project Structure

### LINKED ISSUE #41: Project Structure
Opening this up so we have a place to discuss project structure. I think the structure shown [here](https://www.freecodecamp.org/news/how-to-structure-your-project-and-manage-static-resources-in-react-native-6f4cfc947d92/) could work well for us.

### Project Structure

```
index.js
App.js
ios/
android/
src
  screens
    login
      LoginScreen.js
      LoginNavigator.js
    onboarding
      OnboardingNavigator    
      welcome 
        WelcomeScreen.js
      term
        TermScreen.js
      notification
        NotificationScreen.js
    main
      MainNavigator.js
      news
        NewsScreen.js
      profile
        ProfileScreen.js
      search
        SearchScreen.js
  library
    package.json
    components
      ImageButton.js
      RoundImage.js
    utils
      moveToBottom.js
      safeArea.js
    networking
      API.js
      Auth.js
  res
    package.json
    strings.js
    colors.js
    palette.js --> Rename | This is typography.js in our project
    fonts.js --> Remove | We handle this in typograpy.js
    shapes.js --> Add
    svg.js --> Add
    svg --> Add
    img.js --> Rename | To match svg assets
    img
      logo@2x.png
      logo@3x.png
      button@2x.png
      button@3x.png
scripts
  img.js
  svg.js --> Add | This will make it easier to import svg assets
  clear.js
```

We can also take this opportunity to introduce new practices from that same article that will make our lives easier.

- Use package.json to avoid relative path
- The R namespace (We use index.js to handle this right now but I like this better)
- Scripts for generating resources (I've been doing this manually for svgs and it's a pain)

### PR DESCRIPTION
**Pull request by alfalcon90**
Fixes #41 

- Created new `src` folder on root
- Created three subfolders under `src`: `library`, `res`, and `screens`
- Moved components under `library/components` and misc files under `library/utils` (**let me know if there's a better home for these files**)
- Moved assets (fonts, images, vector files, colors, shapes, and font styles) under `src/res`
- Moved the map and splash views under `src/screens`
- Created package.json files under `library`, `res`, and `screens` to make their content easier to access
- Created new `strings.js` file under `src/res` for all of app copy

### CODE REVIEW

File: src/screens/mapbox/index.js [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+export default class Map extends Component {
+  constructor() {
+    super();
+
+    bindMethods(['goToLocation', 'onRegionDidChange', 'goToLocationNonFirstHelper', 'goToLocationFirstHelper', 'onDidFinishRenderingFrameFully'], this);
+    this.state = {
+      userLocation: null,
+      heading: null,
+      zoomLevel: null,
+      isMapLoading: false, // axiom: loading only applies to an existing map
```
Comment (abdih) [inline src/screens/mapbox/index.js:54]: I don't think this is used, so we can remove it.

File: src/screens/mapbox/index.js [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+      });
+    })
+  }
+
+  goToLocationFirstHelper() {
+    const doUpdates = () => {
+      Geolocation.getCurrentPosition(
+        position => {
+          const {latitude, longitude} = position.coords;
+          console.log('doing first time go to location update');
```
Comment (abdih) [inline src/screens/mapbox/index.js:141]: please remove debugging logs.

File: src/screens/mapbox/index.js [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+      the landscape view here is  e to me not knowing a better alternative to ensure map takes full page size.
+      also, tried adding this as a proper jsx comment next to the respective view, but to no avail.
+    */
+    const {zoomLevel, followUser, haveLocationPermission, goingToLocation, cameraCoordinates, heading} = this.state;
+    return (
+      <View style={styles.landscape}>
+            <MapView
+                animated={true}
+                style={styles.map}
+                styleURL={'mapbox://styles/alfalcon/cka1xbje712931ipd6i5uxam8'}
```
Comment (abdih) [inline src/screens/mapbox/index.js:191]: we can probably extract this string under the `res` folder?

File: src/screens/splash/index.js [hunk 4]
```diff
@@ -1,15 +1,15 @@
...
+import config from 'tinytown/config';
+import { colors, shapes } from 'res'
 
 
 const {RNTwitterSignIn} = NativeModules;
 
 export default class Splash extends Component {
-  constructor(props) {
-    super(props);
+  constructor() {
```
Comment (abdih) [inline src/screens/splash/index.js:11]: 💡 
your IDE is likely telling you that there is a deprecation with the usage around the `constructor` and `super`. this is simply telling you that the `super` should be called with only `props` and nothing else.

you may also be getting a warning related to `props` not being used, but I think this is an exceptional case where even though it's not being used, having it is actually preferred.

File: src/screens/mapbox/index.js [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+  }
+});
+
+const degreeUpdateRate = 3;
+
+const coordinateThreshold = 1 * Math.pow(10, -14);
+
+export default class Map extends Component {
+  constructor() {
+    super();
```
Comment (abdih) [inline src/screens/mapbox/index.js:47]: 💡

**Review by abdih — CHANGES_REQUESTED** [review]

File: src/res/index.js [hunk 6]
```diff
@@ -1,11 +1,13 @@
...
+import * as colors from './colors';
 import * as typography from './typography';
 import * as shapes from './shapes';
+import * as strings from './strings';
 
 export {
   colors,
   typography,
   shapes,
+  strings,
```
Comment (abdih) [inline src/res/index.js:12]: no trailing commas.

File: src/screens/mapbox/index.js [hunk 7]
```diff
@@ -1,12 +1,13 @@
 import React, { Component } from 'react';
 import { StyleSheet, View, PermissionsAndroid, Platform } from 'react-native';
 import StaticSafeAreaInsets from 'react-native-static-safe-area-insets';
-import config from '../../config';
+import config from 'tinytown/config';
```
Comment (abdih) [inline src/screens/mapbox/index.js:4]: we spoke about this earlier, but please make sure to remove the android-related prompt notifying the user that their location will be tracked. thank you.

**Review by abdih — CHANGES_REQUESTED** [review]
minor requests

**Review by keedyc — COMMENTED** [review]
This all looks good to me, @alfalcon90.

Just one question. I see in issue #41 that you were interested in using the R namespace. Is that still of interest?

**Comment by alfalcon90** [comment]
I think we were using the R namespace for something else so I didn't want to break anything but now I can't find it so that might've changed since the mapbox refactor.

**Pull request merged**